File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int n=nums.size();

        int sum=0;
        int maxsum=INT_MIN;

        int sI=0,eI=0;


        for(int i=0;i<n;i++){

            sum+=nums[i];

            if(maxsum<sum){
                maxsum=sum;
                eI=i;
            }
            
            if(sum<0){
                sum=0;
                sI=i+1;
            }
        }

        for(int i=sI;i<=eI;i++) cout<<nums[i]<<" ";
        return maxsum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int sum=0;
        int maxSum=Integer.MIN_VALUE;

        int start=-1;
        int end=-1;
        int ansStart=-1;


        for(int i=0;i<nums.length;i++){
            
            if(sum==0){
                start=i;
            }
            sum+=nums[i];
            if(maxSum<=sum){
                maxSum=sum;
                end=i;
                ansStart=start;
            }
            if(sum<0){
                sum=0;
            }
        }

        //System.out.println(ansStart+","+end);
        for(int i=ansStart;i<=end;i++) System.out.print(nums[i]+" ");
        return maxSum;

    }
}